### ISSUE TITLE
citation-check oracle is nondeterministic and breaks integration oracle gate

### ISSUE THREAD

**Issue by bingran-you** [benchflow-ai/benchflow#772]
## Summary

The live integration oracle determinism gate failed on PR #769 because the citation-check oracle self-scored 0.0. I reproduced the root cause locally: docs/examples/task-md/real-skillsbench/citation-check/oracle/solve.sh calls live CrossRef and Semantic Scholar APIs, and transient lookup failures/rate limits cause real papers to be marked fake.

## Evidence

Command run locally against the checked-in citation-check fixture:

BENCHFLOW_WORKSPACE=/tmp/citation-check-oracle-0RSDr1 bash docs/examples/task-md/real-skillsbench/citation-check/oracle/solve.sh

Observed output contained 7 fake citations instead of the verifier-expected 3. False positives included real papers such as Attention is All You Need, Deep Residual Learning for Image Recognition, BoolQ, and Pre-trained Language Models Can be Fully Zero-Shot Learners. Semantic Scholar rate limiting was also observed during the run.

GitHub CI symptom:

- PR #769 integration-eval failed in tests/test_integration_suite.py::test_oracle_determinism_docker
- Failure: oracle did not self-score 1.0: {citation-check: 0.0}

## Impact

This blocks otherwise unrelated PRs on a nondeterministic external API dependency in a deterministic oracle gate.

## Proposed fix

Make the citation-check oracle deterministic for the checked-in static fixture: parse test.bib and emit the known fake citation titles for the fixture, instead of calling live academic APIs. The task prompt can still require agents to reason about fake citations; the oracle used by CI should be stable ground truth.

**Comment by bingran-you** [comment]
Update 2026-06-15: fixed in PR #773.

Repro confirmed locally:
- Running citation-check/oracle/solve.sh against the checked-in test.bib produced 7 fake citations instead of the verifier-expected 3.
- False positives included real papers such as Attention is All You Need, Deep Residual Learning for Image Recognition, BoolQ, and Pre-trained Language Models Can be Fully Zero-Shot Learners.
- Semantic Scholar rate limiting was observed during the oracle run.
- This matches PR #769 CI failure: test_oracle_determinism_docker reported citation-check reward 0.0.

Fix summary:
- The citation-check oracle is now deterministic for the checked-in static fixture.
- It parses test.bib and emits the known fake citation titles by fixture keys instead of calling live academic APIs.
- Added a fast regression test that runs the oracle and asserts the verifier-expected answer.

Validation completed locally:
- citation-check oracle regression test -> pass
- tests/test_task_check_eval_consistency.py -> 25 passed
- bash -n oracle/solve.sh -> pass
- oracle + verifier smoke -> verifier 9 passed
- uv run python -m pytest tests/ -> 4137 passed, 4 skipped, 7 deselected
- uv run ty check src/ -> pass
- uv run ruff check . -> pass

Local Docker daemon is unavailable, so GitHub CI must confirm the live Docker integration gate.

**Comment by bingran-you** [comment]
Status refresh: PR #773 is now the active fix and is green on GitHub (`test`, `pip-audit`, and `eval-and-judge` passed; merge state CLEAN). I closed older duplicate PR #768 in favor of #773. Remaining gate is human review before merge per repo policy.
